cmd/provisioner-localpv/app/provisioner_blockdevice.go

```diff
@@ -79,7 +83,7 @@ func (p *Provisioner) ProvisionBlockDevice(opts pvController.VolumeOptions, volu
 		WithAccessModes(pvc.Spec.AccessModes).
 		WithVolumeMode(fs).
 		WithCapacityQty(pvc.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]).
-		WithLocalHostDirectory(path).
+		WithFormatLocalHostPath(path, fsType).
```

Comment:
can we have two separate methods?
One that sets host path & other that sets filesystem

Reply:
Thinking aloud -- since the path and format have to go together -- does it still make sense to keep them together. Or we will have to move attaching the VolumeSource into a separate method like: `WithLocalVolumeSourceBuilder().WithPath().WithFSType()`

Reply:
We dont need to have LocalVolumeSourceBuilder as long as LocalVolumeSource contains only a few properties.

I am agreeing to your thought process. Do have a look if below look fine:
```go
WithLocalHostPath(path string)
WithFilesystem(fsType string)
WithLocalHostPathFormat(path, fsType string) // same as yours with name change
WithLocalHostPathAsExt4(path string)
```